### DISCUSSION TITLE
Can we get the output variable returned from send_config_set() if error_pattern is found?

### DISCUSSION THREAD

**Discussion by cvzn1** [ktbyers/netmiko#2802] [Q&A]
Hello,
Is there a way to get the current value of output back once error_pattern is found? The code here in send_config_set() says there isn't:
```
if error_pattern:
    if re.search(error_pattern, output, flags=re.M):
        msg = f"Invalid input detected at command: {cmd}"
        raise ConfigInvalidException(msg)
```
Some applications might want to know what the accumulated output was prior to the error.

If not, what is the recommended way to get this information?

**Comment by ktbyers** [comment]
The exception message tells you the configuration command that failed (so you should know from your list of configuration commands which ones completed successfully).

```
        msg = f"Invalid input detected at command: {cmd}"
```

Is that sufficient?

If not can you expand on your use case, what information you're trying to get?

**Reply by cvzn1** [reply]
We want to render in a textarea (or similar) the command line output for all prior successful commands, then tell the user that execution stopped. The user can confirm what has been configured so far, then choose to continue the configuration if they want, or send a new set. The way it is now, the user just gets an invalid command response, but then has no feedback from the previous commands.

**Reply by ktbyers** [reply]
You should be able to catch the exception and use the `output` variable?

That should have everything that was recorded at that point.

Let me know if that doesn't work for some reason.

@cvzn1

**Reply by cvzn1** [reply]
```
if re.search(error_pattern, output, flags=re.M):
     msg = f"Invalid input detected at command: {cmd}"
     raise ConfigInvalidException(msg)
```
Not sure the exception returns the information contained in the `output` variable?

**Reply by ktbyers** [reply]
Yeah, you're right...okay we would have to add it to the exception msg. Something like:

```
msg = f"""Invalid input detected at command: {cmd}

-------- configuration commands sent before failure --------
{output}
-------------------------------------------------------------
"""
```

I would be open to a PR on something like that.

**Reply by cvzn1** [reply]
Ok great! Thanks for engaging this conversation!

**Reply by ktbyers** [reply]
At least with the above, you could grab the exception message and parse/extract the "output".
